File: LAB_9_10/LAB10 20161811_20161442/LAB10 20161811_20161442/GameSoft-Netbeans/GameSoftController/src/pucp/gamesoft/rrhh/controller/DAOFactory.java
package pucp.gamesoft.rrhh.controller;
import pucp.gamesoft.rrhh.controller.config.DBManager;
import pucp.gamesoft.rrhh.controller.dao.DAODepartment;
import pucp.gamesoft.rrhh.controller.dao.DAOEmployee;

public abstract class DAOFactory {
    public static DAOFactory getDAOFactory(){
        if(DBManager.getDBManager().getUrl().contains("mysql"))
            return new MySQLDAOFactory();
        else
            return new MSSQLDAOFactory();
    }
    public abstract DAOEmployee getEmployeeDAO();
    public abstract DAODepartment getDepartmentDAO();
}

File: LAB_9_10/LAB10 20161811_20161442/LAB10 20161811_20161442/GameSoftLab10CS/GameSoft/GameSoftDAOFactory/DAOFactory.cs
using GameSoftConfig;
using GameSoftDAO;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace GameSoftController
{
    public abstract class DAOFactory
    {
        public static DAOFactory getDAOFactory()
        {
            if (DBManager.DbManager.Type.Contains("mysql"))
                return new MySQLDAOFactory();
            else
                return new MSSQLDAOFactory();
        }
        public abstract DAOProject ProjectDAO { get; }
        public abstract DAOPhaseProject PhaseProjectDAO { get; }
        public abstract DAOActivity ActivityDAO { get; }
        public abstract DAOTask TaskDAO { get; }
        public abstract DAOTime TimeDAO { get; }
        public abstract DAOMoney MoneyDAO { get; }

        public abstract DAOParticipant ParticipantDAO { get; }
    }
}
